src/harness/util.py

```diff
@@ -45,76 +52,67 @@ def _getRandomLatLongInPolygon(polygon):
   next_year_date = datetime.now().replace(year=datetime.now().year + 1)
+  fcc_channel_id = '1'
+  ppa_fips_code = ppa_record['fipsCode']
+  ppa_census_year = ppa_record['censusYear']
+  del ppa_record['censusYear'], ppa_record['fipsCode']
 
-  for pal_record in pal_records:
+  for index, pal_rec in enumerate(pal_records):
+    pal_rec = defaultdict(lambda: defaultdict(dict), pal_rec)
     # Change the FIPS Code and Registration Date-Year in Pal Id
-    pal_record['palId'] = '/'.join([str(ppa_fips_code) if index == 2 else
-                           '%s-%d' % ('{:02d}'.format(previous_year_date.month),
-                                      previous_year_date.year)
-                           if index == 1 else str(val) for index, val in
-                                    enumerate(pal_record['palId'].split('/'))])
-    pal_record['userId'] = device['userId']
+    pal_rec['palId'] = '/'.join(['pal', '%s-%d' % ('{:02d}'.format(previous_year_date.month),
+                                                   previous_year_date.year), str(ppa_fips_code),
+                                 fcc_channel_id])
+    pal_rec['userId'] = user_id
     # Make the date consistent in Pal Record for Registration and License
-    pal_record = _changeDateInPpaAndPal(pal_record,
-                                        previous_year_date.strftime('%Y-%m-%dT%H:%M:%SZ'),
-                                        next_year_date.strftime('%Y-%m-%dT%H:%M:%SZ'))
+    pal_rec['registrationInformation']['registrationDate'] = \
+      previous_year_date.strftime('%Y-%m-%dT%H:%M:%SZ')
+
     # Change License Information in Pal
-    pal_record['license']['licenseAreaIdentifier'] = str(ppa_fips_code)
-    pal_record['license']['licenseAreaExtent'] = \
+    pal_rec['license']['licenseAreaIdentifier'] = str(ppa_fips_code)
+    pal_rec['license']['licenseAreaExtent'] = \
       'zone/census_tract/census/%d/%d' % (ppa_census_year, ppa_fips_code)
+    pal_rec['license']['licenseDate'] = previous_year_date.strftime('%Y-%m-%dT%H:%M:%SZ')
+    pal_rec['license']['licenseExpiration'] = next_year_date.strftime('%Y-%m-%dT%H:%M:%SZ')
+    pal_rec['license']['licenseFrequencyChannelId'] = fcc_channel_id
     # Change Frequency Information in Pal
-    for assign in ('primaryAssignment', 'secondaryAssignment'):
-      pal_record['channelAssignment'][assign]['lowFrequency'] = low_frequency
-      pal_record['channelAssignment'][assign]['highFrequency'] = high_frequency
-  # Make the date consistent in Ppa Record
-  ppa_record = _changeDateInPpaAndPal(ppa_record,
-                                      previous_year_date.strftime('%Y-%m-%dT%H:%M:%SZ'),
-                                      next_year_date.strftime('%Y-%m-%dT%H:%M:%SZ'))
+    pal_rec['channelAssignment']['primaryAssignment']['lowFrequency'] = low_frequency
+    pal_rec['channelAssignment']['primaryAssignment']['highFrequency'] = high_frequency
+    pal_records[index] = json.loads(json.dumps(pal_rec))
```

Comment:
Thanks for that suggestion! I think the defaultdict is not required for PPA Record because we are making changes in the root level. But in the case of PAL Record the addition of key has the nesting that's why I have used defaultdict over there.